src/satellite_images_nso_datascience/model_metrics/custom_model_metrics.py

```diff
@@ -0,0 +1,296 @@
+            and self.satellite_constellation == "Superview"
+        ):
+            # TODO: FIll with more difficult pixels.
+            fault_check.append(
+                [
+                    "Water",
+                    self.model.predict(
+                        self.scaler.transform([[275, 208, 117, 219, 97, 88]])
+                    )
+                    == "Water",
+                ]
+            )
+
+            fault_check.append(
+                [
+                    "Water",
+                    self.model.predict([[1024, 656, 575, 416, 122, 57]]) == "Water",
+                ]
+            )
+
+        if len(fault_check) == 0:
+            return (
+                "Error no difficult pixels found for the location or the constellation."
+            )
+
+        return fault_check
+
+    def metrics_on_small_tif_files(self):
+        """
+        Test the model on small random .tif files, one side for model performance on the other side for the connection with raw .tif files.
+
+        """
+
+        # Collect the small .tif files to make predictions on.
+        if (
+            self.satellite_constellation == "Superview"
+            and self.location == "Nieuwkoopse_plassen"
+        ):
+            collect_files = glob.glob(
+                os.path.abspath("./tests/test_input/").replace("\\", "/")
+                + "/Nieuwkoopse_plassen_raster_images/*SV*.tif"
+            )
+        elif (
+            self.satellite_constellation == "PNEO"
+            and self.location == "Nieuwkoopse_plassen"
+        ):
+            collect_files = glob.glob(
+                os.path.abspath("./tests/test_input/").replace("\\", "/")
+                + "/Nieuwkoopse_plassen_raster_images/*PNEO*.tif"
+            )
+
+        return_metrics = []
+
+        for afile in collect_files:
+            afile = afile.replace("\\", "/")
+
+            df = self.__raster_to_dataframe(afile)
+
+            df["filename"] = afile
+            df["date"] = afile.split("/")[-1][0:15]
+            df["label"] = afile.split("_test")[0].split("_")[-1]
+
+            # Get the most common prediction file and base that as it's predicted value.
+            predict_value = (
+                pd.Series(
+                    self.model.predict(
+                        self.scaler.transform(df[self.selected_features])
+                    )
+                )
+                .value_counts()
+                .index[0]
+            )
+
+            return_metrics.append(
+                [
+                    afile,
+                    predict_value == afile.split("_test")[0].split("_")[-1],
+                    predict_value,
+                    afile.split("_test")[0].split("_")[-1],
+                ]
+            )
+
+        return return_metrics
+
+    def metrics_on_small_tif_file_iterator(self):
+        """
+        Test the model performance on custom .tif files but also to test if  the model performance well with the iterator.
```

Comment:
You describe 2 tests here model performance and if the code works well with the iterator. Didn't we discuss that we wanted to split those 2 types of testing, metrics vs unit tests. I'm confused as to why you put them together anyway.

Reply:
Might be worth discussing but with time issue's i would suggest this for a other pr

Reply:
Okay then.